Name: New Card Button In Browser
Version: 1.3.5-1
Author: Jason Robitaille
Description: Adds a new card button in the browser that opens up new browser cards  (MIT license)

--- .orig/usr/palm/applications/com.palm.app.browser/app/controllers/page-navigationmenu.js
+++ /usr/palm/applications/com.palm.app.browser/app/controllers/page-navigationmenu.js
@@ -22,6 +22,12 @@
 		command: 'stop'
 	},
 	
+	NewCard: {
+		label: $L('New Card'),
+		icon: 'new',
+		command: 'new-page-cmd'
+	},
+	
 	Reload: {
 		label: $L('Reload'),
 		icon: 'refresh',
--- .orig/usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js
+++ /usr/palm/applications/com.palm.app.browser/app/controllers/page-controls.js
@@ -25,6 +25,13 @@
 			icon: MenuData.NavigationMenu.Forward.icon,
 			command: MenuData.NavigationMenu.Forward.command
 		};
+		
+		this._newcardModel = {
+			disabled: false,
+			label: MenuData.NavigationMenu.NewCard.label,
+			icon: MenuData.NavigationMenu.NewCard.icon,
+			command: MenuData.NavigationMenu.NewCard.command
+		};
 
 		this._model = {
 			visible: true,
@@ -48,6 +55,7 @@
 			this._model.items.push(idleModel);
 		}
 
+		this._model.items.push(this._newcardModel);
 		this._controller.setupWidget(Mojo.Menu.commandMenu, {menuClass: 'no-fade'}, this._model);
 	},
 
@@ -74,6 +82,7 @@
 
 		// The spacer... right aligns the following button.
 		items.push({});
+		items.push(this._newcardModel);
 
 		this._back = back;
 		this._forward = forward;
